tfhe/benches/core_crypto/pbs_bench.rs

```diff
@@ -0,0 +1,184 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use tfhe::boolean::parameters::{BooleanParameters, DEFAULT_PARAMETERS, TFHE_LIB_PARAMETERS};
+use tfhe::core_crypto::prelude::*;
+use tfhe::shortint::parameters::*;
+use tfhe::shortint::Parameters;
+
+macro_rules! named_param {
+    ($param:ident) => {
+        (stringify!($param), $param)
+    };
+}
```

Comment:
Try using the already implemented NamedParam trait from

tfhe::shortint::keycache

then you should be able to call e.g. PARAM_MESSAGE_2_CARRY_2.name()

Reply:
Sadly no since parameters with `.*_CARRY_0` have no parameter name and thus panic when we call `name()` from `NamedParam` trait.

Reply:
we can add the implementation for the parameters missing the impl

Reply:
Resolved by adding the parameters used in the corresponding macro.